packages/cli/src/e2e.test.ts

```diff
@@ -0,0 +1,179 @@
+import { configFileSchema } from './validators/schemas';
+import { cli } from './cli';
+import { randomUUID } from 'crypto';
+
+describe('e2e', () => {
+    const out = 'generated/synthql/e2e/tests';
+    const configFile = 'src/tests/synthql.config.json';
+
+    beforeAll(() => {
+        fs.rmSync(out, {
+            force: true,
+            recursive: true,
+        });
+    });
+
+    beforeEach(() => {
+        vitest.resetModules();
+        vitest.resetAllMocks();
+    });
+
+    test('Write schema to files', async () => {
+        const prettierOptions = await prettier.resolveConfig(
+            '../../.prettier.config.js',
+        );
+
+        expect(prettierOptions).not.toBe(null);
+
+        // Generate in `cli` package
+        const cliOutDir = '../cli/src/tests';
+
+        if (!fs.existsSync(cliOutDir)) {
+            fs.mkdirSync(cliOutDir, { recursive: true });
+        }
+
+        fs.writeFileSync(
+            path.join(cliOutDir, 'synthql.dev.config.json'),
+            await prettier.format(JSON.stringify(configFileSchema, null, 2), {
+                parser: 'json',
+                ...prettierOptions,
+            }),
+        );
+
+        // Generate in `docs` package
+        const docsOutDir = '../docs/static/schemas';
+
+        if (!fs.existsSync(docsOutDir)) {
+            fs.mkdirSync(docsOutDir, { recursive: true });
+        }
+
+        fs.writeFileSync(
+            path.join(docsOutDir, 'synthql.config.json'),
+            await prettier.format(JSON.stringify(configFileSchema, null, 2), {
+                parser: 'json',
+                ...prettierOptions,
+            }),
+        );
+    }, 100_000);
+
+    test(
+        `Generate schema types with a supplied config file option, --configFile`,
+        async () => {
+            const outDir = `${out}/${Date.now()}-${randomUUID()}`;
+
+            const argv = [
+                'node',
+                'generate-with-default-schema.js',
+                'generate',
+                `--configFile=${configFile}`,
+                `--out=${outDir}`,
+            ];
+
+            // Compose file path
+            const outFilePath = path.join(outDir, `db.ts`);
+
+            // Check that file with the name doesn't exist yet
+            expect(!fs.existsSync(outFilePath)).toEqual(true);
+
+            // Generate schema file
+            const output = await cli({ argv, exit: () => {} });
+
+            // Check that the file now exists
+            expect(fs.existsSync(outFilePath)).toEqual(true);
+
+            // Verify that the option data is correct
+            expect(output.configFile).toEqual(configFile);
+
+            // Verify that the passed attribute overrides were applied
+            const schemaFilePath = path.join(outDir, `schema.ts`);
+
+            const file = await import(schemaFilePath);
+            const schema = file.schema.properties;
+
+            const schemaDefOverrides = output.schemaDefOverrides as any;
```

Comment:
Please don't use `any`. Is there a reason why you need it here?

Reply:
Indeed. Same reason why we needed the` ajv `typing. `schemaDefOverrides` is always `unknown`. I thought of using the validator function (`validateSchemaDefOverrides`)to validate the value, but that would be adding a dependency to the tests. The idea of using any here is that we want this test to fail with a reference error, the moment the data shape is not what it expects, without having to use a function from a dependency

Reply:
I think using the validator makes a lot of sense, specially given that this is an e2e test (we're testing the whole flow, including validation). Why would adding a dependency on it be a bad idea?

Reply:
The way I was thinking about it, since we were using the validator to parse the data inside the CLI function, it seemed like testing the output of the validator process using the same validator... but if we're okay with that, then it makes sense to use it